lib/commands/startup.rb

```diff
@@ -0,0 +1,73 @@
+require 'colorize'
+require 'net/http'
+require 'timeout'
+
+require_relative './base'
+require_relative './run'
+
+class Commands::Startup < Commands::Base
+  def call(variation = nil)
+    stack = variation ? "app-#{variation}" : "app"
+
+    print_live_hostname
+
+    Commands::Run
+      .new(config_directory: config_directory, service: service, stack: stack, verbose: verbose)
+      .call
+  end
+
+private
+
+  def print_live_hostname
+    return unless hostname
+
+    Thread.new do
+      url = "http://#{hostname}"
+
+      Timeout::timeout(30) do
+        wait_until_can_visit?(url)
+
+        puts
+        puts "Application is available at: #{url}".blue
+        puts
+      end
+    rescue Timeout::TimeoutError
+      puts "Warning: Unable to communicate with application within 30 seconds."
+    end
+  end
+
+  def possible_hostnames
+    path = File.join(config_directory, "services", "nginx-proxy", "docker-compose.yml")
+    docker_compose = YAML.load_file(path)
+    docker_compose["services"]["nginx-proxy-app"]["networks"]["default"]["aliases"]
+  end
+
+  def find_hostname
+    search_hostname = "#{service.tr('_-', '')}.dev.gov.uk"
+    possible_hostnames.each do |hostname|
+      return hostname if hostname.tr('_-', '') == search_hostname
```

Comment:
What are we guarding against with the `.tr` here and on line 45? Is it just because we inevitable forget which services have underscores and which have dashes when we type?

Reply:
At the moment it's only `govuk_publishing_components` that needs this, https://github.com/alphagov/govuk-docker/pull/50.

The service is `govuk_publishing_components`, the address is `govuk-publishing-components.dev.gov.uk`.